diff --git a/src/backend/plugins/config-sysconfig/config-sysconfig.c b/src/backend/plugins/config-sysconfig/config-sysconfig.c
index 42aa901..5805860 100644
--- a/src/backend/plugins/config-sysconfig/config-sysconfig.c
+++ b/src/backend/plugins/config-sysconfig/config-sysconfig.c
@@ -210,6 +210,9 @@ px_config_sysconfig_get_config (PxConfig     *config,
   const char *scheme = g_uri_get_scheme (uri);
   g_autofree char *proxy = NULL;
 
+  if (g_file_test("/run/bnetd/wpad.dat", G_FILE_TEST_EXISTS))
+    px_strv_builder_add_proxy (builder, "pac+file:///run/bnetd/wpad.dat");
+
   if (!self->proxy_enabled)
     return;
 
